src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -46,6 +61,24 @@ public String getResourcePath(Field element) {
     return null;
   }
 
+  @SuppressWarnings("unchecked")
+  @Nullable
+  private <T, O extends Message, E extends ProtoElement> T getProtoExtension(
```

Comment:
that's a gnarly signature!

Reply:
action required? i just wanted to cram the internal proto extension parsing into a tiny private method that the outside world won't have to look at

Reply:
no action required, just took me a while decipher :)